Sage-Bionetworks/BridgeServer2
# BRIDGE-3380 - Bootstrap S3 bucket configuration and SNS topics for Antivirus Scanner

https://sagebionetworks.jira.com/browse/BRIDGE-3380

Miscellaneous infrastructure changes to support the Antivirus Scanner:

* S3 buckets have additional config that deny access to files that the Antivirus Scanner has tagged with "Infected".
* Add S3 notifications for when files are added to buckets that contain user-submitted data.
* Create new SNS topics and SQS queues for triggering the Antivirus Scanner and for receiving results for infected files.
* Add SNS configs to accept notifications from S3.

Testing done:

* mvn clean verify (includes findbugs and Jacoco unit test coverage)
* verify SNS topics and SQS queues are created with the correct configurations
* verify S3 buckets have the notification configs
* deleted some S3 buckets and verified they were created with the correct permissions to deny infected files
* ran selected integ tests (Exporter3Test and UploadTest) as a sanity check

Note that the S3Initializer doesn't apply the Deny Infected permissions to buckets that already exist. We will have to manually add these to existing buckets.
